src/fee-calculator.ts

```diff
@@ -26,34 +37,84 @@ export class FeeCalculator {
     );
     if (!balanceData) return;
 
-    // Get first date from balance data
-    const balanceDates = Object.keys(balanceData.balances).sort();
-    if (balanceDates.length === 0) return;
+    // Get all dates from balance data and sort chronologically
+    const sortedDates = Object.keys(balanceData.balances).sort();
+    if (sortedDates.length === 0) return;
 
-    const firstDate = balanceDates[0]!;
-    const firstBalance = balanceData.balances[firstDate]!;
+    // Process all dates to create daily entries
+    const dailyEntries = this.createDailyEntries(
+      sortedDates,
+      balanceData.balances,
+    );
 
-    // Create fee report
+    // Create and write fee report
     const feeReport: FeeReport = {
       metadata: {
         chain_id: CHAIN_IDS.ARBITRUM_NOVA,
```

Comment:
This should always be fetched from the provider, unless it's already present in the file. never use a hardcoded value for chain id